### PR TITLE
Additional changes to windows terrain

### PR DESCRIPTION
**Pull request by TheGoatGod**
Summary
`SUMMARY: Content "Additional changes to windows terrain"`

Purpose of change
there were a few things out of place

Describe the solution
I fixed windows allowing you to be able to see through them when the curtain was closed, added an extra metal grate window to allow building on no glass windows and framed windows. fixed up skylight frame there 1 entry instead of 8 there all still there. moved windows category because I feel it should be above other and lastly I fixed some lower case spelling to upper case.

Describe alternatives you've considered
don't merge

Testing
I have tested and everything I changed is how it should be

### CODE REVIEW

**Comment by TheGoatGod** [comment]
not sure why there is rakes of commits but when I compared there wasn't any conflicts or file changes

File: data/json/construction.json [hunk 1]
```diff
@@ -3601,11 +3601,39 @@
...
+    "components": [ [ [ "pipe", 12 ] ], [ [ "sheet_metal", 4 ] ] ],
+    "pre_terrain": "t_window_reinforced_noglass",
+    "post_terrain": "t_metal_grate_window"
+  },
   {
     "type": "construction",
     "id": "constr_metal_grate_window_curtain",
     "description": "Build Metal Grate Over a Window",
-    "//": "Step 2: Build curtain over metal grate with curtain",
+    "//": "Step 3: Build curtain over metal grate with curtain",
```
Comment (Night-Pryanik) [inline data/json/construction.json:3622]:
```suggestion
    "//": "Step 3: Build curtain over metal grate",
```

File: data/json/construction.json [hunk 2]
```diff
@@ -3601,11 +3601,39 @@
     "pre_terrain": "t_window_reinforced",
     "post_terrain": "t_metal_grate_window"
   },
+  {
+    "type": "construction",
+    "id": "constr_metal_grate_window_no_glass",
+    "description": "Build Metal Grate Over a Window With No Glass",
```
Comment (Night-Pryanik) [inline data/json/construction.json:3607]:
```suggestion
    "description": "Build Metal Grate Over a Window Without Glass",
```

File: data/json/construction.json [hunk 3]
```diff
@@ -3601,11 +3601,39 @@
...
+    "tools": [ [ [ "oxy_torch", 4 ], [ "welder", 20 ], [ "welder_crude", 30 ], [ "toolset", 30 ] ] ],
+    "qualities": [ [ { "id": "SAW_M", "level": 1 } ], [ { "id": "GLARE", "level": 2 } ] ],
+    "components": [ [ [ "nail", 4 ] ], [ [ "sheet", 2 ] ], [ [ "stick", 1 ] ], [ [ "string_36", 1 ] ] ],
+    "pre_terrain": "t_metal_grate_window",
+    "post_terrain": "t_metal_grate_window_with_curtain"
+  },
+  {
+    "type": "construction",
+    "id": "constr_metal_grate_window_curtain_no_glass",
+    "description": "Build Metal Grate Over a Window With No Glass",
```
Comment (Night-Pryanik) [inline data/json/construction.json:3635]:
```suggestion
    "description": "Build Metal Grate Over a Window Without Glass",
```

File: data/json/construction.json [hunk 4]
```diff
@@ -3601,11 +3601,39 @@
...
+    "//": "Step 4: Build curtain over metal grate with a curtain",
```
Comment (Night-Pryanik) [inline data/json/construction.json:3636]:
```suggestion
    "//": "Step 4: Build curtain over metal grate",
```

**Comment by Night-Pryanik** [comment]
As you add `curtains` examine action to many new types of windows, you need to update this examine action in `iexamine.cpp` - there is a hardcoded set of types of windows which will change to another types of windows after tearing down the curtains.

**Comment by TheGoatGod** [comment]
I don't know what I'm doing in that kinda code.

**Pull request merged**